_stbt/core.py

```diff
@@ -1244,8 +1249,18 @@ def wait_until(callable_, timeout_secs=10, interval_secs=0, predicate=None,
 
     while True:
         t = time.time()
-        value = callable_()
-        predicate_value = predicate(value)
+        try:
+            value = callable_()
```

Comment:
Could this be

```
try:
    value = callable_()
    predicate_value = predicate(value)
except ...
```

and then you could avoid the `else:` block and the second `try...except`?

Unless this has something to do with not setting `value = None` if predicate fails?